irods/manager/data_object_manager.py

```diff
@@ -75,6 +75,8 @@ def put(self, file, irods_path, **options):
             options[kw.UPDATE_REPL_KW] = ''
             self.replicate(obj, **options)
 
+        return self.get(obj)
```

Comment:
I have concerns about potential performance issues here (mostly thinking about automated ingest) as `get` runs at least 2 queries (line 45 and 54).

Would it be horrible to add a new entry to the `options` dictionary to indicate that we want a reference to the new data object to be returned? It could be used for the other interfaces as well (e.g. `create`, `register`, etc.). I realize this would just transfer the work from making a `get` call on your part to setting some keyword. Perhaps the default could be to return the `iRODSDataObject` and you must set some option to prevent it for performance.

Thoughts? Any other ideas (or an explanation as to why this wouldn't be a bad performance hit) are welcome!

Reply:
Another option might be to implement a `find()` method that can be called at the end of the `put()` method (and the others you mentioned), that returns a reference only. I'm ok with whichever default you'd prefer here, be it returning the reference or not, and using a switch in the options dict to indicate the other choice.

Reply:
What keyword should be used for this? I'm not seeing anything in `keywords.py` that looks promising, nor is there anything in the upstream `rodsKeyWdDef.h`. Do those two files have to match?

Reply:
I think the optional argument seems fine. My thought with the keywords was that since `**options` is just a dict, we could just add a check for some new key existing with value `True` before returning. It "shouldn't" affect iRODS (could be wrong about that - like I said, could be a horrible idea). We would be hijacking `**options`, in a way, but this way wouldn't affect the interface. In any case, I'm fine with either of these options.

As for the question about `rodsKeyWdDef.h`, I think the idea is that there should be matching keywords in each file (although `keywords.py` may be incomplete).

Thanks!